Sources/Swiftfall/Swiftfall.swift

```diff
@@ -1,635 +1,637 @@
-        public let cmc:Int?
-
-        // The type line of this card.
-        public let type_line:String?
-        
-        // The Oracle text for this card, if any.
-        public let oracle_text:String?
-       
-        // The mana cost for this card. This value will be any empty string "" if the cost is absent. Remember that per the game rules, a missing mana cost and a mana cost of {0} are different values.
-        public let mana_cost:String?
-        
-        // This card’s power, if any. Note that some cards have powers that are not numeric, such as *.
-        public let power: String?
-        
-        // This card’s toughness, if any. Note that some cards have toughnesses that are not numeric, such as *.
-        public let toughness: String?
-       
-        // This loyalty if any. Note that some cards have loyalties that are not numeric, such as X.
-        public let loyalty: String?
-        
-        // This card’s colors.
-        public let colors:[String]?
-        
-        // Online listings for these cards names.
-        public let purchase_uris: [String:String]
-    
-        // Flavor text on the card, if there is any
-        public let flavor_text: String?
-        
-        // id of the illustration
-        public let illustration_id: String?
-        
-        // uris of the images
-        public let image_uris:[String:String]?
-        
-        public func simplePrint(){
-            // if the card has multiple faces, print them
-            if (self.card_faces) != nil {
-                for face in card_faces! {
-                    face.simplePrint()
-                }
-                return
-            } 
-            
-            // Each variable is tested to see if printing it makes sense.
-            if self.name != nil {
-                print("Name: \(name!)")
-            }
-            if self.mana_cost != nil {
-                print("Cost: \(mana_cost!)")
-            }
-            if self.type_line != nil {
-                print("Type Line: \(type_line!)")
-            }
-            if self.oracle_text != nil {
-                print("Oracle Text:\n\(oracle_text!)")
-            }
-            if self.power != nil && self.toughness != nil {
-                print("Power: \(power!)\nToughness: \(toughness!)")
-            }
-            if self.loyalty != nil {
-                print("Loyalty: \(loyalty!)")
-            }
-            
-        }
-    }
-    
-    // fuzzy
-    public static func getCard(fuzzy: String) -> Card?
-    {
-        let encodeFuzz = fuzzy.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
-        let call = "cards/named?fuzzy=\(encodeFuzz)"
-
-        var card: Card?
-        var stop = false
-        parseCard(call: call){
-            (newcard:Card?) in
-            card = newcard
-            stop = true
-        }
-        
-        while(!stop){
```

Comment:
This is, unfortunately, not a safe way to do this. When parseCard sets `stop = true`, it does so on a background thread. It is undefined behavior to read from and write to the `stop` variable on different threads.

The way to do this is to use a `DispatchSemaphore`, like so:

```swift
var card: Card?
let semaphore = DispatchSemaphore(value: 0)

parseCard(call: call) { newCard in
  card = newCard
  semaphore.signal()
}

semaphore.wait() // This will sit and wait until `semaphore.signal()` is called on the other thread.
```

For more information: https://medium.com/swiftly-swift/a-quick-look-at-semaphores-6b7b85233ddb